### PR TITLE
fix typescript and moved api to reusable files

### PR DESCRIPTION
**Pull request by mountclear**
I've moved the the existing FE APIs that calls the backend to their own individual file. 
This is so it can be used elsewhere if need. 

I also move the Typescript files to the adjacent fetch files. 
This is done as the return of the response we want to define the type that response is as this will eventually be consumes by the other files.

### CODE REVIEW

**Comment by mountclear** [comment]
Also can you test this out please just so I have not broken other parts I may have missed anything else.

**Comment by mountclear** [comment]
Also WTF is this:
Screenshot 2023-06-25 at 01 45 56

**Comment by mountclear** [comment]
Also I converted the component SearchBox to a normal funciton, but I can't see this being used so hopefully I've not broken anything as it's not being used I believe.

File: frontend/api/fetchLifterNames/fetchLifterNames.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { LifterSearchList } from "./fetchLifterNamesTypes"
+
+export default async function fetchLifterNames(nameQuery: string): Promise<LifterSearchList> {
+  if (nameQuery.length < 3) {
+    return {names: []}
+  }
+
+  try {
+    const response = await fetch(`https://api.openweightlifting.org/search?name=${nameQuery}`);
```
Comment (euanwm) [inline frontend/api/fetchLifterNames/fetchLifterNames.ts:9]: Reference the process.env.API variable like in the other fetch / call functions.

Comment (mountclear) [reply frontend/api/fetchLifterNames/fetchLifterNames.ts:9]: ok changed thought it was a different domain intentionally so left it

**Comment by euanwm** [comment]
Tested and it seems to all be OK. @mountclear

**Pull request merged**